symfony/plugins/orangehrmSecurityAuthenticationPlugin/lib/dao/SecurityAuthenticationDao.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+class SecurityAuthenticationDao extends BaseDao {
+
+    /**
+     * @param ResetPasswordLog $resetPasswordLog
+     * @return bool
+     * @throws DaoException
+     */
+    public function saveResetPasswordLog(ResetPasswordLog $resetPasswordLog) {
+
+        try {
+            if ($resetPasswordLog instanceof ResetPasswordLog) {
+
+                $resetPasswordLog->save();
+            }
+            return true;
+        } catch (Exception $e) {
+
+            throw new DaoException($e->getMessage());
+        }
+    }
+
+    /**
+     * @param $email
+     * @return array|Doctrine_Record
+     * @throws DaoException
+     */
+    public function getResetPasswordLogByEmail($email) {
+
+        try {
+            $q = Doctrine_Query::create()
+                ->from("ResetPasswordLog")
+                ->where('reset_email = ?', $email)
+                ->andWhere('status = ?', 0);
+
+            $resetPasswordLog = $q->fetchOne();
+            return $resetPasswordLog;
+        } catch (Exception $e) {
+            throw new DaoException($e->getMessage());
+        }
+    }
+
+    /**
+     * @param $username
+     * @param $newPassword
+     * @return Doctrine_Collection
+     * @throws DaoException
+     */
+    public function saveNewPrimaryPassword($username, $newPassword) {
+        try {
+            $query = Doctrine_Query::create()
+                ->update('SystemUser')
+                ->set('user_password', '?', $newPassword)
```

Comment:
ahh .. this is wrong .. you can't save plain password in the db .. Use SystemUser Service to save the password

Reply:
I encrypt password in savePassword() function on SecurityAuthenticationService
using  $primaryHash = $this->getSystemUserService()->hashPassword($newPrimaryPassword);
function. In this case we are not save plain text